pom.xml

```diff
@@ -172,13 +184,27 @@
         <version>${project.version}</version>
       </dependency>
 
+      <!-- Use latest Guava -->
+      <dependency>
+        <groupId>com.google.guava</groupId>
+        <artifactId>guava</artifactId>
+        <version>33.3.0-jre</version>
```

Comment:
This is a transitive dependency of `compile-testing`, to make sure that we don't accidentally get Guava on the classpath I would add:
```xml
<scope>test</scope>
```

there should be a very good reason to change it ...

Reply:
I suspect this might not work

Reply:
We already do it with other dependencies: https://github.com/dylibso/chicory/blob/264f57dbada145dc43d1c8ebc458a81e2857e43c/pom.xml#L220

Reply:
This turns out not to work due to Guava being a dependency of Jimfs, which is used by `Wast2Json` during the invocation of `TestGenMojo`. Thus, from the perspective of `test-gen-plugin`, it is not a test dependency, so the scope shouldn't be changed.

Reply:
not a big deal, but I would overwrite the scope in the `test-gen-plugin` to clearly indicate that is an expected "exception".
But fine with anything ...

Reply:
Setting the scope in dependency management and then overriding it when necessary seems error prone.

As long as we review any non-test dependencies added to the core modules, we shouldn’t need to worry about things being added accidentally. The sorting based on scope makes it easy to see. They currently have no dependencies and that seems to be a project goal, so it should be easy to review.

Reply:
Note that `test-gen-plugin` doesn’t depend on Guava directly. It’s a transitive dependency. So we’d have to add it as a dependency just to override the scope.

Reply:
> currently have no dependencies and that seems to be a project goal, so it should be easy to review

It's definitely a project goal! Fine with it :+1: